be/test/vec/columns/common_column_test.h

```diff
@@ -0,0 +1,766 @@
+// for example column_array should test this function:
+// size, reserve, resize, empty, byte_size, allocated_bytes, clone_resized,
+// get_shrinked_column, filter, filter_by_selector, serialize_vec, deserialize_vec, get_max_row_byte_size
+//
+namespace doris::vectorized {
+
+class CommonColumnTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        col_str = ColumnString::create();
+        col_str->insert_data("aaa", 3);
+        col_str->insert_data("bb", 2);
+        col_str->insert_data("cccc", 4);
+
+        col_int = ColumnInt64::create();
+        col_int->insert_value(1);
+        col_int->insert_value(2);
+        col_int->insert_value(3);
+
+        col_dcm = ColumnDecimal64::create(0, 3);
+        col_dcm->insert_value(1.23);
+        col_dcm->insert_value(4.56);
+        col_dcm->insert_value(7.89);
+
+        col_arr = ColumnArray::create(ColumnInt64::create(), ColumnArray::ColumnOffsets::create());
+        Array array1 = {1, 2, 3};
+        Array array2 = {4};
+        col_arr->insert(array1);
+        col_arr->insert(Array());
+        col_arr->insert(array2);
+
+        col_map = ColumnMap::create(ColumnString::create(), ColumnInt64::create(),
+                                    ColumnArray::ColumnOffsets::create());
+        Array k1 = {"a", "b", "c"};
+        Array v1 = {1, 2, 3};
+        Array k2 = {"d"};
+        Array v2 = {4};
+        Array a = Array();
+        Map map1, map2, map3;
+        map1.push_back(k1);
+        map1.push_back(v1);
+        col_map->insert(map1);
+        map3.push_back(a);
+        map3.push_back(a);
+        col_map->insert(map3);
+        map2.push_back(k2);
+        map2.push_back(v2);
+        col_map->insert(map2);
+    }
+
+    ColumnString::MutablePtr col_str;
+    ColumnInt64::MutablePtr col_int;
+    ColumnDecimal64::MutablePtr col_dcm;
+    ColumnArray::MutablePtr col_arr;
+    ColumnMap::MutablePtr col_map;
+
+    void checkColumn(const IColumn& col1, const IColumn& col2, const IDataType& dataType,
+                     size_t column_size) {
+        if (WhichDataType(dataType).is_map()) {
+            auto map1 = check_and_get_column<ColumnMap>(col1);
+            auto map2 = check_and_get_column<ColumnMap>(col2);
+            const DataTypeMap& rhs_map = static_cast<const DataTypeMap&>(dataType);
+            checkColumn(map1->get_keys(), map2->get_keys(), *rhs_map.get_key_type(),
+                        map1->get_keys().size());
+            checkColumn(map2->get_values(), map2->get_values(), *rhs_map.get_value_type(),
+                        map1->get_values().size());
+        } else {
+            if (WhichDataType(dataType).is_int8()) {
+                auto c1 = check_and_get_column<ColumnInt8>(col1);
+                auto c2 = check_and_get_column<ColumnInt8>(col2);
+                for (size_t i = 0; i < column_size; ++i) {
+                    EXPECT_EQ(c1->get_element(i), c2->get_element(i));
```

Comment:
I think you can use get() or [] to get field and check filed1 == field2 for all types.